File: examples/libvetest8.f90
function mod_buff(arr, res, n)
    integer, dimension(n), intent(in) :: arr
    integer, dimension(n), intent(out) :: res
    integer i, mod_buff
    mod_buff = 0
    do i = 1, n
       mod_buff = mod_buff + arr(i)
       res(i) = arr(i) + 2
    end do
    !print *, arr
    !print *, "ftn finished."
    return
end

File: examples/libvetest8.c
#include <stdio.h>

int mod_buff(int *a, int *b, int *c)
{
	int i, sum = 0, n = *c;

	printf("VE: &a=%x &b=%x n=%d\n", (void *)a, (void *)b, n);
        for (i = 0; i < n; i++) {
		printf("VE:(%d) a[%d]=%d\n", i, i, a[i]);
		sum += a[i];
		b[i] = a[i] + 2;
	}
	return sum;
}

